README.md

```diff
@@ -2,5 +2,435 @@
+        blockingJdbcContext = newFixedThreadPool(databaseConfig.maxPoolSize.value()).asCoroutineDispatcher()
+    )
+    val queryExecutor = JdbcQueryExecutor(transactionManager)
+    val dslContext: DSLContext = DSL.using(
+        POSTGRES,
+        Settings().withRenderNamedParamPrefix("$").withParamType(ParamType.NAMED)
+    )
+
+    /**
+     * Persisting definition
+     */
+    val tracer = NoopTracerFactory.create()
+    val walletRepo = WalletRepository(queryExecutor, dslContext)
+    val persisting = Persisting(
+        transactionManager = transactionManager,
+        modelRepos = ModelRepos(Wallet::class hasRepo walletRepo),
+        eventRepository = JooqEventRepository(queryExecutor, dslContext, tracer)
+    )
+
+    /**
+     * Unit of work executor definition
+     */
+    val clock = Clock.tickMillis(UTC)
+    val uowx: UnitOfWorkExecutor = UnitOfWorkExecutor(
+        persisting = persisting,
+        tracer = tracer,
+        meterRegistry = SimpleMeterRegistry(),
+        factories = listOf(
+            CreateWalletUow::class withFactory { CreateWalletUow(walletRepo, clock) }
+        )
+    )
+}
+```
+
+### Run it!
+> Please don't forget to create tables for your models and table for storing events.
+You can find script to create event's table [here](eva-events-db-schema/src/main/resources/com/razz/eva/events/db)
+
+```kotlin
+    val module = WalletModule(config)
+    val principal = ServicePrincipal(Principal.Id("eva-id"), Principal.Name("eva"))
+    
+    val createdWallet = module.uowx.execute(CreateWalletUow::class, principal) {
+        CreateWalletUow.Params(
+            id = "45dfd599-4d62-47f1-8e47-a779df4f6bbc",
+            currency = "USD"
+        )
+    }
+```
+
+## Features
+
+### Event sourcing
+...
+
+### Unit of work validation
+After you wrote your first unit of work, you probably want to ask - how I can test it?
+
+We provide verification DSL, so you can write unit tests and verify results of your unit of work.
+Use `verifyInOrder` function to start verification process.
+```kotlin
+    CreateWalletUow(queries, clock).tryPerform(principal, params) verifyInOrder {
+        adds<Wallet> { model -> ... }
+        addsEq(expectedModel)
+        
+        updates<Wallet> { model -> ... }
+        updatesEq(expectedModel)
+
+        emits<WalletEvent> { event -> ... }
+        emitsEq(expectedEvent)
+    
+        returns { result -> ... }
+    }
+```
+You can check some examples [here](eva-uow/src/test/kotlin/com/razz/eva/uow/UnitOfWorkDemoSpec.kt)
+
+### Idempotency
+Sometimes something goes wrong, your service doesn't respond within deadline. You want to make a retry, but you are afraid of creating duplicates or new unnecessary models, so your DB becomes inconsistent.
+
+To prevent it people use [idempotency key](https://stripe.com/docs/api/idempotent_requests) pattern.
+Unit of work allows you to define idempotency key in params, so you can safely make retries.
```

Comment:
It won't work out of the box, we need to provide example of handling idempotency conflicts in uow (outer code).

Reply:
Also, implementation of idempotent execution is tied to events table and need to be explained.